Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantAssetBuilder.cpp

```diff
@@ -1177,16 +1177,18 @@ namespace AZ
                     AZStd::vector<AZStd::string> fullCommand;
                     fullCommand.push_back(command);
                     AZStd::string failMessage;
-                    if (!LaunchRadeonGPUAnalyzer(fullCommand, creationContext.m_tempDirPath, failMessage))
+                    if (LaunchRadeonGPUAnalyzer(fullCommand, creationContext.m_tempDirPath, failMessage))
                     {
-                        return AZ::Failure(failMessage);
+                        // add rga output to the by product list
+                        outputByproducts->m_intermediatePaths.insert(AZStd::string::format(
+                            "./%s_disassem_%u_frag.txt", shaderVariantInfo.m_asic.c_str(), shaderVariantInfo.m_stableId));
+                        outputByproducts->m_intermediatePaths.insert(AZStd::string::format(
+                            "./%s_livereg_%u_frag.txt", shaderVariantInfo.m_asic.c_str(), shaderVariantInfo.m_stableId));
+                    }
+                    else
+                    {
+                        AZ_Warning(ShaderVariantAssetBuilderName, false, failMessage.c_str());
```

Comment:
When reporting strings, we should also pass through "%s" just in case the string has "%" symbols that could become a stability or security issue.
```suggestion
                        AZ_Warning(ShaderVariantAssetBuilderName, false, "%s", failMessage.c_str());
```